TP/Frontend/src/app/components/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,149 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { VentasService } from '../../services/ventas.service';
+import { ProductCardsService } from 'src/app/services/product-cards.service';
+import { UserService } from 'src/app/services/user.service';
+import { Producto } from '../../model/productos';
+import { MatDialog } from '@angular/material/dialog';
+import { DialogoComponent } from '../dialogo/dialogo.component';
+
+declare var M: any;
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss'],
+})
+export class ProductDetailComponent implements OnInit {
+  idProducto: string;
+  CarrouselElems: any;
+  CarrouselInstance: any;
+  producto = new Producto();
+  vendedor: any = {};
+  usuario: any = {};
+  vendedorIsNotComprador: any;
+  imagenVendedor: string =
+    'https://res.cloudinary.com/elcurco8/image/upload/v1598910919/TTADS-TP/user_ybrhuc.png';
+
+  constructor(
+    private route: ActivatedRoute,
+    private ventas: VentasService,
+    private pService: ProductCardsService,
+    private userService: UserService,
+    private vService: UserService,
+    private router: Router,
+    public dialogo: MatDialog
+  ) {}
+
+  ngOnInit(): void {
+    //el usuario es el que tengo en el localStorage
+    this.usuario = this.userService.getLocalUser();
+
+    // me traigo el id de Producto
+    this.idProducto = this.route.snapshot.paramMap.get('idProducto');
+    //me traigo el producto
+    this.pService.getProducto(this.idProducto).subscribe((res: Producto) => {
```

Comment:
evitemos las abreviaturas este caso es muy nitido, pService, vService, si los miro rapido parecen lo mismo 
productService sellerService es mucho mas claro.

Reply:
algunos consejos sobre variables https://readlearncode.com/code-and-stuff/clean-code-variables/